pennylane_lightning/src/tests/Test_GateImplementations_Param.cpp

```diff
@@ -0,0 +1,1480 @@
+#include "TestHelpers.hpp"
+#include "TestKernels.hpp"
+#include "Util.hpp"
+
+#include <catch2/catch.hpp>
+
+#include <algorithm>
+#include <complex>
+#include <iostream>
+#include <limits>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+/**
+ * @file This file contains tests for parameterized gates. List of such gates is
+ * [RX, RY, RZ, PhaseShift, Rot, ControlledPhaseShift, CRX, CRY, CRZ, CRot]
+ */
+
+using namespace Pennylane;
+/**
+ * @brief Run test suit only when the gate is defined
+ */
+#define PENNYLANE_RUN_TEST(GATE_NAME)                                          \
```

Comment:
Thanks for the suggestion! Still, it means we have a file with ~2000 lines of code, which I want to avoid. I think we can anyway make some automated tests if the compile-time JSON works (discussed today in Slack). Can we revisit this issue then?